model/cells/zhong_cell.hpp

```diff
@@ -30,69 +30,98 @@ class zhong_cell : public cell<T, std::string, sir, vicinity> {
 
     using config_type = simulation_configuration;
 
-    using PhaseRates = std::vector<float>;
+    using phase_rates = std::vector<            // The age sub_division
+                        std::vector<double>>;   // The stage of infection
 
-    std::vector<PhaseRates> virulence_rates;
-    std::vector<PhaseRates> recovery_rates;
-    std::vector<PhaseRates> mobility_rates;
+    phase_rates virulence_rates;
+    phase_rates recovery_rates;
+    phase_rates mobility_rates;
+
+    double disobedient;  // percentage of population that do not follow the quarantine restrictions
+
+    // To make the parameters of the correction_factors variable more obvious
+    using infection_threshold = float;
+    using mobility_correction_factor = float;
+
+    std::map<infection_threshold, mobility_correction_factor> correction_factors;
 
-    float disobedient;
     int precDivider;
 
     zhong_cell() : cell<T, std::string, sir, vicinity>() {}
 
-    zhong_cell(std::string const &cell_id, cell_unordered<vicinity> const &neighborhood, sir initial_state, std::string const &delay_id, simulation_configuration config) :
+    zhong_cell(std::string const &cell_id, cell_unordered<vicinity> const &neighborhood, sir initial_state,
+            std::string const &delay_id, simulation_configuration config) :
     cell<T, std::string, sir, vicinity>(cell_id, neighborhood, initial_state, delay_id) {
 
         virulence_rates = std::move(config.virulence_rates);
         recovery_rates = std::move(config.recovery_rates);
         mobility_rates = std::move(config.mobility_rates);
 
+        correction_factors = std::move(config.correction_factors);
         disobedient = config.disobedient;
+
         precDivider = config.precision;
 
-        assert(virulence_rates.size() == recovery_rates.size() &&
-                       virulence_rates.size() == mobility_rates.size() &&
-               "\n\nThere must be an equal number of phases between all configuration rates.\n\n");
+        assert(virulence_rates.size() == recovery_rates.size() && virulence_rates.size() == mobility_rates.size() &&
+               "\n\nThere must be an equal number of age segments between all configuration rates.\n\n");
     }
 
     // user must define this function. It returns the next cell state and its corresponding timeout
     sir local_computation() const override {
```

Comment:
We will have to re-adapt this function to the new approach (shouldn't be too much work, is following Sebastian's way)